torchtext/experimental/transforms.py

```diff
@@ -144,22 +147,169 @@ class TextSequentialTransforms(nn.Sequential):
+
+        Examples:
+            >>> txt_pipeline('the pretrained sp model names')
+        """
         for module in self:
-            input = module(input)
-        return input
+            line = module(line)
+        return line
 
     def to_ivalue(self):
-        r"""Return a JITable TextSequentialTransforms.
+        r"""Call to_ivalue func if any modules have the func.
         """
         module_list = []
         for _idx, _module in enumerate(self):
             if hasattr(_module, 'to_ivalue'):
                 _module = _module.to_ivalue()
             module_list.append((str(_idx), _module))
         return TextSequentialTransforms(OrderedDict(module_list))
+
+
+_pretrained_spm = ['text_unigram_15000', 'text_unigram_25000', 'text_unigram_50000',
+                   'text_bpe_15000', 'text_bpe_25000', 'text_bpe_50000']
+
+
+class PretrainedSPTokenizer(nn.Module):
+    r"""Tokenizer based on a pretained sentencepiece model
+
+    Args:
+       spm_model: the pretrained sentencepiece model names. Default: 'text_unigram_25000'. The following pretrained sentencepiece models are provided:
+            - text_unigram_15000
+            - text_unigram_25000
+            - text_unigram_50000
+            - text_bpe_15000
+            - text_bpe_25000
+            - text_bpe_50000
+            Otherwise, the file path to the user-provided sentencepiece model is required.
+
+    Examples:
+        >>> import torch
+        >>> from torchtext.experimental.transforms import PretrainedSPTokenizer
+        >>> spm_tokenizer = PretrainedSPTokenizer('text_unigram_25000')
+        >>> spm_tokenizer = PretrainedSPTokenizer('.data/text_unigram_25000.model')
+        >>> jit_spm_tokenizer = torch.jit.script(spm_tokenizer)
+    """
+
+    def __init__(self, spm_model='text_unigram_25000'):
+        super(PretrainedSPTokenizer, self).__init__()
+        if spm_model in _pretrained_spm:
+            spm_model = download_from_url('https://pytorch.s3.amazonaws.com/models/text/pretrained_spm/{}.model'.format(spm_model))
+        try:
+            self.sp_model = load_sp_model(spm_model)
+        except:
+            raise RuntimeError('The pretrained sentencepiece model is not supported')
+
+    def forward(self, line: str) -> List[str]:
+        r"""
+        Args:
+            line: the input string
+
+        Examples:
+            >>> spm_tokenizer('the pretrained sp model names')
+            >>> ['▁the', '▁pre', 'trained', '▁sp', '▁model', '▁names']
+        """
+
+        return self.sp_model.EncodeAsPieces(line)
+
+    @torch.jit.export
+    def decode(self, tokens: List[str]) -> str:
+        r"""
+        Args:
+            tokens: the tokens list for decoder
+
+        Examples:
+            >>> spm_transform.decoder(['▁the', '▁pre', 'trained', '▁sp', '▁model', '▁names'])
+            >>> 'the pretrained sp model names'
+        """
+
+        return self.sp_model.DecodePieces(tokens)
+
+
+class PretrainedSPTransform(nn.Module):
```

Comment:
Same thing here. Also what are these model pretrained on? what's the dataset, what do the numbers mean, are there going to be more? what are the settings used and can the user get them somewhere?

Reply:
The spm provided here was trained by `bookcorpus`, `enwik9`, `wiki103` datasets. I will update the doc strings to reflect this.